test/integration/vaultdynamicsecret_integration_test.go

```diff
@@ -339,14 +342,19 @@ func TestVaultDynamicSecret(t *testing.T) {
 					))
 					require.Equal(t, 1, len(pods.Items))
 					assert.Equal(t, pods.Items[0].UID, vdsObjFinal.Status.LastRuntimePodUID)
+					assert.Equal(t, vdsObjFinal.GetGeneration(), vdsObjFinal.Status.LastGeneration)
```

Comment:
I think this indirectly is a duplicate of line 331 since we are not doing any `Get()` of the `vdsObjFinal` between line 331 and here, so these will always be true if line 331 is true.